eng/scripts/build.ps1

```diff
@@ -11,6 +11,11 @@ function Process-Sdk ($path) {
         go clean -v ./...
     }
 
+    if ($clearOutput) {
+        Write-Host "##[command]Cleaning auto-generated files in" $path
+        rm "zz_generated_*"
```

Comment:
Only some of these commands actually seem to use `$path`, others (like this and `go clean` just use the current directory. I assume this is a bug?

Reply:
I actually do not know what the command `go clean` does here. I would defer @chamons to answer this and discuss whether we need that option.

I agree we should unify the usage of path here. But actually I do not like the way that we are passing the absolute path everywhere - how about we just remove the $path parameter from this function and make sure this function only takes effects in whatever the current directory is. The caller should ensure we are in the correct directory first, and then invoke this function, like this script does.

Reply:
I'm realizing `$path` is probably more for being able to parse pipeline logs and understand which context things failed in, though that could probably also be achieved by logging directory changes when they happen.

I think the only consideration with absolute pathing is when you're running scripts to work against multiple project directories, error handling/control flow becomes a little simpler because you won't get dropped into subdirectories accidentally and you don't have to handle pushd/popd all over the place. Additionally if the scripts ever need to reference calls to other scripts/tools, then having one consistent script directory is more convenient (though I think just using `$PSScriptRoot` may solve that problem).

Reply:
The case here is that based on my understanding of how the go tools work (which might not be correct), some of them should either accept a relative directory (go vet for instance), or a full package identifier which is not necessary to be an absolute path. Therefore I prefer to use the relative path in the invocations of commands.
I added another variable to first get the absolute path of current directory once we go into the function so that we would have that in our verbose logging.